src/parametric/services/ParametricUtils.jl

```diff
@@ -294,34 +310,32 @@ function GraphSolveBuffers(@nospecialize(M), ::Type{T}) where {T}
   return GraphSolveBuffers(ϵ, p, X, Xc)
 end
 
-struct GraphSolveContainer
+struct GraphSolveContainer{CFT}
   M::AbstractManifold # ProductManifold or ProductGroup
   buffers::OrderedDict{DataType, GraphSolveBuffers}
   varTypes::Vector{DataType}
   varTypesIds::OrderedDict{DataType, Vector{Symbol}}
-  cfdict::OrderedDict{Symbol, CalcFactorMahalanobis}
   varOrderDict::OrderedDict{Symbol, Tuple{Int, Vararg{Int}}}
-  # cfarr::AbstractVector # TODO maybe <: AbstractVector(CalcFactorMahalanobis)
+  cfv::ArrayPartition{CalcFactorMahalanobis, CFT}
```

Comment:
```suggestion
  cfv::ArrayPartition{<:CalcFactorMahalanobis, CFT}
```

Reply:
The ArrayPartition should be able to store a mixture of any CalcFactor type, so the `<:` will not work here.

Reply:
I'm not actually sure if the first parameter is used at all. 
I'm thinking of moving the optim.jl solve to the attic in any case.